packages/story-editor/src/components/canvas/canvasLayout.js

```diff
@@ -77,12 +88,13 @@ function CanvasLayout({ header }) {
         <CanvasUploadDropTarget>
           <CanvasElementDropzone>
             <SelectionCanvas>
-              <DisplayLayer />
-              <FramesLayer />
+              {hasDimensions && <DisplayLayer />}
+              {hasDimensions && <FramesLayer />}
               <NavLayer header={header} />
             </SelectionCanvas>
             <EditLayer />
             <EyedropperLayer />
+            {hasDimensions && <VideoCaptionsLayer />}
```

Comment:
Curious why not display it inside the display layer? Maybe it would be more straightforward to keep the captions inside the visible area when using the display layer (e.g. in case of zoomed videos).

Also, I'm wondering what should happen with the captions if the video is partially out of the page, should the captions be forced to the visible area only?

Reply:
I'll look into moving this to the DisplayLayer, haven't tested that at all.

> Also, I'm wondering what should happen with the captions if the video is partially out of the page, should the captions be forced to the visible area only?

The video captions position should never change, regardless of video position.

Reply:
Moved it to DisplayLayer now, not sure that makes a difference or brings an advantage. Seems kinda pointless since the captions position should never change anyway.

Reply:
Hm, I suppose not -- when I first tested it, I actually saw the captions on top of the video (in the bottom area), so I assumed the idea was to display the captions on top of the video that it relates to, but on top of all other elements... That's why I thought that maybe having it in the `DisplayLayer` of that specific element might be beneficial. But it looks like it's actually displayed in the same place related to the page, misunderstood that. Since the display layer includes everything that's related to what the user gets displayed, should still be good to have it there though.

What happens if there are more than one video with captions playing at the same time?

Reply:
> What happens if there are more than one video with captions playing at the same time?

On the frontend the captions are stacked.

In the editor it's not even possible to play 2 videos at the same time, as only the currently selected video can be played.

Reply:
Not directly related to this PR but it seems to me that when a video has captions, the element links on that page don't work anymore in the FE. Is it just me or can you also reproduce it? The `captions-area` seems to cover everything and prevent the links from being clicked.

Reply:
Yikes, yeah I can reproduce it. Opened #9819 for this. Good catch!

Reply:
One more thing I'm wondering about -- it looks like currently, using a Page Attachment and Video captions on the same page is also problematic (the captions get partially hidden under the Page Attachment) -- is this expected behaviour?

Reply:
Do you mean in the editor or on the frontend?

I just saw that on the frontend it can happen only on smaller screens:

![Screenshot 2021-11-26 at 11 25 08](https://user-images.githubusercontent.com/841956/143567022-784ed456-fe33-4740-9c83-15d850ede04b.png)

Not sure how easy (and urgent) it would be to change the position for smaller screens.

On the frontend you'll notice that the page attachment is positioned at the very bottom of the page:

![Screenshot 2021-11-26 at 11 31 26](https://user-images.githubusercontent.com/841956/143567199-da2f377a-9c99-4391-a821-42157b284d13.png)

Whereas in the editor, the page attachment has lots of space below it:
![Screenshot 2021-11-26 at 11 26 36](https://user-images.githubusercontent.com/841956/143567308-71e1c0db-1cd8-4005-b5e6-e5563e929bdc.png)

So that will lead to captions getting partially hidden:

![Screenshot 2021-11-26 at 11 33 46](https://user-images.githubusercontent.com/841956/143567388-3cd6a081-485f-45a8-b00e-30e8955e5d79.png)

I think the proper solution here is to move the page attachment further down in the editor, so it matches what you actually see on the frontend.

Reply:
> Do you mean in the editor or on the frontend?

> I just saw that on the frontend it can happen only on smaller screens:


I meant in the front-end. I'm currently using a 13'' screen so that's a smaller screen and explains why I saw it right away, wasn't aware it works as expected on larger screens.

Reply:
cc @barklund related to the Page Attachment positioning in the editor since you're currently working on #7566 (in case it's easy to adjust it as part of that issue in Fit mode as well)

Reply:
@miina Is this resolved now with #7566?

Reply:
Should be, the Page Attachment is now fixed close to the bottom edge as in the front-end, however, on smaller screens these (captions + the link) still overlap.